File: net/schedule-unpublishing/AddFilter.cs
// DocSection: schedule_unpublishing_add_filter
// Tip: Find more about .NET SDKs at https://docs.kontent.ai/net
using Kentico.Kontent.Delivery;

// Initializes a content delivery client
IDeliveryClient client = DeliveryClientBuilder
      .WithProjectId("8d20758c-d74c-4f59-ae04-ee928c0816b7")
      .Build();

// Gets all articles
// Create strongly typed models according to https://docs.kontent.ai/strongly-typed-models
var response = await client.GetItemsAsync<Article>(
    new EqualsFilter("system.type", "article")
);

var today = System.DateTime.Today;

// Filters the articles, keeping those that should be public
var itemsToDisplay = response.Items.Where((item) =>
		(item.PublishUntil > today || item.PublishUntil == null));

return View(itemsToDisplay);
// EndDocSection

File: android/schedule-unpublishing/AddFilter.java
// DocSection: schedule_unpublishing_add_filter
// Tip: Find more about Java/JavaRx SDKs at https://docs.kontent.ai/javaandroid
import com.github.kentico.kontent_delivery_core.*;
import com.github.kentico.kontent_delivery_rx.*;

import io.reactivex.Observer;
import io.reactivex.disposables.Disposable;
import io.reactivex.functions.Function;

Date now = new Date();

// Prepares an array to hold strongly-typed models
List<TypeResolver<?>> typeResolvers = new ArrayList<>();

// Registers the type resolvers
typeResolvers.add(new TypeResolver<>(Article.TYPE, new Function<Void, Article>() {
    @Override
    public Article apply(Void input) {
        return new Article();
    }
}));

// Prepares the DeliveryService configuration object
String projectId = "8d20758c-d74c-4f59-ae04-ee928c0816b7";
IDeliveryConfig config = DeliveryConfig.newConfig(projectId)
    .withTypeResolvers(typeResolvers);

// Initializes a DeliveryService for Java projects
IDeliveryService deliveryService = new DeliveryService(config);

// Gets all articles using a simple request
List<Article> articles = deliveryService.<Article>items()
        .equalsFilter("system.type", "article")
        .get()
        .getItems();

List<Article> publishedItems = new ArrayList<>();

for (Article article : articles) {
    if (
        article.getPublishUntil() == null || article.getPublishUntil().after(now)) {
        publishedItems.add(article);
    }
}

// Gets all articles using RxJava2
deliveryService.<Article>items()
        .equalsFilter("system.type", "article")
        .getObservable()
        .subscribe(new Observer<DeliveryItemListingResponse<Article>>() {
            @Override
            public void onSubscribe(Disposable d) {
            }

            @Override
            public void onNext(DeliveryItemListingResponse<Article> response) {
                Date now = new Date();

                // Gets the retrieved articles
                List<Article> articles = response.getItems();

                List<Article> publishedItems = new ArrayList<>();

              	// Filters the articles, keeping those that should be public
                for (Article article : articles) {
                    if (article.getPublishUntil() == null || article.getPublishUntil().after(now)) {
                        publishedItems.add(article);
                    }
                }
            }

            @Override
            public void onError(Throwable e) {
            }

            @Override
            public void onComplete() {
            }
        });
// EndDocSection